web-common/src/features/chat/core/messages/Messages.svelte

```diff
@@ -29,6 +29,18 @@
   // Data
   $: messages = $getConversationQuery.data?.conversation?.messages ?? [];
 
+  // Build a map of result messages by parent ID for correlation with calls (excluding router_agent)
+  $: resultMessagesByParentId = new Map(
+    messages
+      .filter((msg) => msg.type === "result" && msg.tool !== "router_agent")
```

Comment:
We should move type and tool to enums. Feels weird to see hardcoded values everywhere.